File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/metamodel/hashmap/aas/qualifier/qualifiable/Qualifier.java
package org.eclipse.basyx.aas.metamodel.hashmap.aas.qualifier.qualifiable;

import org.eclipse.basyx.aas.api.metamodel.aas.qualifier.qualifiable.IQualifier;
import org.eclipse.basyx.aas.api.metamodel.aas.reference.IReference;
import org.eclipse.basyx.aas.metamodel.facades.HasSemanticsFacade;
import org.eclipse.basyx.aas.metamodel.facades.QualifierFacade;
import org.eclipse.basyx.aas.metamodel.hashmap.aas.qualifier.HasSemantics;
import org.eclipse.basyx.aas.metamodel.hashmap.aas.reference.Reference;

/**
 * Qualifier class
 * 
 * @author kuhn
 *
 */
public class Qualifier extends Constraint implements IQualifier {

	/**
	 * Version of serialized instances
	 */
	private static final long serialVersionUID = 1L;
	
	public static final String QUALIFIER ="qualifier";
	
	public static final String QUALIFIERTYPE ="qualifierType";
	
	public static final String QUALIFIERVALUE ="qualifierValue";
	
	public static final String QUALIFIERVALUEID="qualifierValueId";

	/**
	 * Constructor
	 */
	public Qualifier() {
		// Add all attributes from HasSemantics
		this.putAll(new HasSemantics());

		// Default values
		put(QUALIFIERTYPE, "");
		put(QUALIFIERVALUE, null);
		put(QUALIFIERVALUEID, null);
	}

	public Qualifier(String type, String value, Reference valueId) {
		// Add all attributes from HasSemantics
		this.putAll(new HasSemantics());

		// Default values
		put(QUALIFIERTYPE, "");
		put(QUALIFIERVALUE, null);
		put(QUALIFIERVALUEID, null);
	}

	@Override
	public void setQualifierType(String obj) {
		new QualifierFacade(this).setQualifierType(obj);
		
	}

	@Override
	public String getQualifierType() {
	return new QualifierFacade(this).getQualifierType();
	}

	@Override
	public void setQualifierValue(Object obj) {
		new QualifierFacade(this).setQualifierValue(obj);
		
	}

	@Override
	public Object getQualifierValue() {
		return new QualifierFacade(this).getQualifierValue();
	}

	@Override
	public void setQualifierValueId(IReference obj) {
		new QualifierFacade(this).setQualifierValueId(obj);
		
	}

	@Override
	public IReference getQualifierValueId() {
		return new QualifierFacade(this).getQualifierValueId();
	}
	
	@Override
	public IReference getSemanticId() {
		return new HasSemanticsFacade(this).getSemanticId();
	}

	@Override
	public void setSemanticID(IReference ref) {
		new HasSemanticsFacade(this).setSemanticID(ref);
		
	}
}

File: sdks/c++/basys.sdk.cc/src/aas/impl/metamodel/hashmap/aas/qualifier/qualifiable/Qualifier.cpp
/*
 * Qualifier.cpp
 *
 *      Author: wendel
 */

#include "Qualifier.h"

namespace basyx {
namespace submodel {
namespace metamodel {
namespace map {
namespace qualifier {
namespace qualifiable {

Qualifier::Qualifier()
{}

Qualifier::Qualifier(const std::string & qualifierType, const basyx::any & qualifierValue, const std::shared_ptr<aas::reference::IReference> & valueId) :
  qualifierType {qualifierType},
  qualifierValue {qualifierValue},
  qualifierValueId {valueId}
{}

std::string Qualifier::getQualifierType() const
{
  return this->qualifierType;
}

basyx::any Qualifier::getQualifierValue() const
{
  return this->qualifierValue;
}

std::shared_ptr<aas::reference::IReference> Qualifier::getQualifierValueId() const
{
  return this->qualifierValueId;
}

std::shared_ptr<aas::reference::IReference> Qualifier::getSemanticId() const
{
  return this->semanticId;
}

void Qualifier::setQualifierType(const std::string & qualifierType)
{
  this->qualifierType = qualifierType;
}

void Qualifier::setQualifierValue(const basyx::any & qualifierValue)
{
  this->qualifierValue = qualifierValue;
}

void Qualifier::setQualifierValueId(const std::shared_ptr<aas::reference::IReference> & valueId)
{
  this->qualifierValueId = valueId;
}

void Qualifier::setSemanticId(const std::shared_ptr<aas::reference::IReference>& semanticId)
{
  this->semanticId = semanticId;
}


}
}
}
}
}
}
